hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -612,7 +612,7 @@ private ValidationResult validate(
         try {
             transactionKeys = allKeysForTransaction(txBody, payer);
         } catch (PreCheckException e) {
-            return new ValidationResult(PRE_HANDLE_FAILURE, KEY_NOT_PROVIDED);
+            return new ValidationResult(PRE_HANDLE_FAILURE, e.responseCode());
```

Comment:
Maybe we can directly use here the logic we need from allKeysForTransaction, since we will be double dispatching         `dispatcher.dispatchPureChecks(nestedTxn); ` as its called earlier in the flow.